internal/pkg/source/source_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"testing"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/analysistest"
+	"golang.org/x/tools/go/analysis/passes/buildssa"
+	"golang.org/x/tools/go/ssa"
+)
+
+type testConfig struct {
+	propagatorsPattern string
+	fieldsPattern      string
+	sanitizerPattern   string
+}
+
+func (c *testConfig) IsSanitizer(call *ssa.Call) bool {
+	match, _ := regexp.MatchString(c.sanitizerPattern, call.String())
+	return match
+}
+
+func (c *testConfig) IsPropagator(call *ssa.Call) bool {
+	match, _ := regexp.MatchString(c.propagatorsPattern, call.String())
+	return match
+}
+
+func (c *testConfig) IsSourceFieldAddr(field *ssa.FieldAddr) bool {
+	match, _ := regexp.MatchString(c.fieldsPattern, field.String())
+	return match
+}
+
+type analyzerResult struct {
+	allocations []*ssa.Alloc
+	calls       []*ssa.Call
+	fieldAddr   []*ssa.FieldAddr
+	store       []*ssa.Store
+}
+
+var testAnalyzer = &analysis.Analyzer{
+	Name:       "source",
+	Run:        run,
+	Doc:        "test harness for the logic related to sources",
+	Requires:   []*analysis.Analyzer{buildssa.Analyzer},
+	ResultType: reflect.TypeOf(analyzerResult{}),
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	in := pass.ResultOf[buildssa.Analyzer].(*buildssa.SSA)
+	var result analyzerResult
+	for _, fn := range in.SrcFuncs {
+		fmt.Printf("Func: %v\n", fn.Name())
```

Comment:
I would prefer these to be `t.Logf` statements so they don't clutter up the command-line `go test ./...`  Although to do that, you'd have to do some closure tricks and reset `testAnalyzer.Run` at the start of every test.

If you want to keep the prints, I'd still prefer the closure.  Otherwise, if it was just for debugging, please remove.

Reply:
Removed.